src/Framework/Framework/Compilation/Binding/GeneralBindingPropertyResolvers.cs

```diff
@@ -429,5 +430,47 @@ public IsMoreThanZeroBindingProperty IsMoreThanZero(ParsedExpressionBindingPrope
                 Expression.GreaterThan(expr.Expression, Expression.Constant(0))
             ));
         }
+
+        public static ReferencedViewModelPropertiesBindingProperty GetReferencedViewModelProperties(IValueBinding binding, ParsedExpressionBindingProperty expression)
+        {
+            var allProperties = new List<PropertyInfo>();
+            var expr = expression.Expression;
+
+            expr.ForEachMember(m => {
+                if (m is PropertyInfo property)
+                {
+                    allProperties.Add(property);
+                }
+            });
+
+            while (true)
+            {
+                // unwrap type conversions, negations, ...
+                if (expr is UnaryExpression unary)
+                    expr = unary.Operand;
+                // unwrap some method invocations
+                else if (expr is MethodCallExpression boxCall && boxCall.Method.DeclaringType == typeof(BoxingUtils))
+                    expr = boxCall.Arguments.First();
+                else if (expr is MethodCallExpression { Method.Name: nameof(DateTimeExtensions.ToBrowserLocalTime) } dtMethodCall && dtMethodCall.Method.DeclaringType == typeof(DateTimeExtensions))
+                    expr = dtMethodCall.Object ?? dtMethodCall.Arguments.First();
+                else if (expr is MethodCallExpression { Method.Name: nameof(object.ToString) } toStringMethodCall)
+                    expr = toStringMethodCall.Object ?? toStringMethodCall.Arguments.First();
```

Comment:
I'd also unwrap the ToEnumString method which we now use for enum->string implicit conversion by default